app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt

```diff
@@ -474,6 +483,25 @@ private fun Fact.toEvent(): Event? = when (Pair(component, item)) {
 
         null
     }
+    Component.BROWSER_AWESOMEBAR to BrowserAwesomeBarFacts.Items.PROVIDER_DURATION -> {
+        metadata?.get(BrowserAwesomeBarFacts.MetadataKeys.DURATION_PAIR)?.let { providerTiming ->
+            require(providerTiming is Pair<*, *>) { "Expected providerTiming to be a Pair" }
+            when (val provider = providerTiming.first as AwesomeBar.SuggestionProvider) {
+                is HistoryStorageSuggestionProvider -> PerfAwesomebar.historySuggestions
+                is BookmarksStorageSuggestionProvider -> PerfAwesomebar.bookmarkSuggestions
+                is SessionSuggestionProvider -> PerfAwesomebar.sessionSuggestions
+                is SearchSuggestionProvider -> PerfAwesomebar.searchEgnineSuggestions
+                is ClipboardSuggestionProvider -> PerfAwesomebar.clipboardSuggestions
+                is ShortcutsSuggestionProvider -> PerfAwesomebar.shortcutsSuggestions
+                // NB: add PerfAwesomebar.syncedTabsSuggestions once we're using SyncedTabsSuggestionProvider
+                else -> {
+                    Logger("Metrics").error("Unknown suggestion provider: $provider")
+                    null
+                }
+            }?.setRawNanos(providerTiming.second as Long)
```

Comment:
A `timespan` metric type can only be set once per ping: subsequent measurements will overwrite prior measurements. I think you'll need to use the `timing_distribution` metric type to record these values: https://mozilla.github.io/glean/book/user/metrics/timing_distribution.html

Reply:
Thanks for the catch!